chore: add 'Reserve company name' form schema

phrogwrld opened the pull request:
## Description
<!-- Summarize the changes based on added/changed functionality --><!-- Summarize the changes based on added/changed functionality -->
This PR adds the form scehma for 'Reserve a Company Name'

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## Changes Made

<!--List the files changed and why -->

### Notes
<!--Add notes for anything unrelated to the specified categories -->

## Testing
- [ ] Visual regression tests added for all device sizes
- [ ] Verify all pages render correctly
- [ ] Test responsive layout across device sizes (snapshots created)
- [ ] Check accessibility standards are met
- [ ] Validate markdown formatting renders properly
- [ ] Test navigation and breadcrumbs

## Related Github Issue(s)/Trello Ticket(s)
<!-- Link any related issues: Fixes #123 -->


## Checklist
- [ ] Code follows project style guidelines
- [ ] Self-review completed
- [ ] Tests added/updated (visual regression snapshots)
- [ ] Documentation updated

phrogwrld committed:
chore: add form for `reserve a company name`

phrogwrld committed:
fix: render fieldArray type in DynamicField for non-conditional fields

sajclarke commented on src/schema/reserve-company-name.ts:
```diff
@@ -0,0 +1,262 @@
+    description: "",
+    fields: [
+      {
+        name: "What is the present name of the company",
```
```suggestion
        name: "companyName",
```
We don't want to use spaces for the field names

sajclarke commented on src/schema/reserve-company-name.ts:
```diff
@@ -0,0 +1,262 @@
+    description: "",
+    fields: [
+      {
+        name: "aNewCompany",
```
```suggestion
        name: "purposeOfCompany",
```

sajclarke commented on src/schema/reserve-company-name.ts:
```diff
@@ -0,0 +1,262 @@
+        validation: { required: "First choice is required" },
+      },
+      {
+        name: "secondChoice",
```
```suggestion
        name: "companyName.secondChoice",
```

phrogwrld committed:
chore: field name revisions

phrogwrld committed:
chore: add entry and start pages

sajclarke commented on src/schema/reserve-company-name.ts:
```diff
@@ -0,0 +1,260 @@
+          pattern: {
+            value: "^\\d{1,2}\\s?\\d{3}\\s?\\d{3}\\s?\\d{4}$",
+            message:
+              "Please enter a valid phone number (e.g., 246 234 5678 or 1 246 234 5678)",
```
```suggestion
              "Please enter a valid phone number (e.g., 1 246 234 5678)",
```

sajclarke reviewed (changes requested):
@phrogwrld please see comments

phrogwrld committed:
Apply suggestions from code review

Co-authored-by: Shannon Clarke <shannonajclarke@gmail.com>

phrogwrld committed:
fix: initialize fieldArray with correct minItems count

Previously, fieldArray fields were always initialized with a single item regardless of the minItems configuration. This caused users to manually add items to reach the minimum before the remove button would appear.

sajclarke committed:
Merge branch 'main' into chore/reserve-company-name
